src/coreclr/tools/aot/ILCompiler.Trimming.Tests/TestCasesRunner/TrimmingArgumentBuilder.cs

```diff
@@ -147,6 +147,7 @@ public virtual void AddStripLinkAttributes (bool stripLinkAttributes)
 
 		public virtual void AddSubstitutions (string file)
 		{
+			Options.SubstitutionFiles.Add (file);
```

Comment:
Is this for #88647? Is it intentionally part of this PR?

Reply:
I guess it would fix that bug as well (at least partially). 
It is intentionally part of this PR - the tests for warn-as-error in illink use substitutions files to generate the warnings (don't ask my why, I don't know why we chose to do that). I wanted to reuse the tests to get a 1:1 validation against trimmer behavior. And this does that. If it fixes other things... why not :-)

I do agree that it's a but unfortunate, but it felt easier than coming up with a separate test for this.